packages/content/lib/database.js

```diff
@@ -46,7 +46,7 @@ class Database extends Hookable {
   query (path, { deep = false, text = false } = {}) {
     const isDir = !path || !!this.dirs.find(dir => dir === path)
     // Look for dir or path
-    const query = isDir ? { dir: deep ? { $regex: new RegExp(`^${path}`) } : path } : { path }
+    const query = isDir ? { dir: deep ? { $regex: new RegExp(`^${path.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')}`) } : path } : { path }
```

Comment:
JFYI: 

Lodash has such function that escapes a regular expression: `escapeRegExp`.
This would be more [well-tested](https://github.com/lodash/lodash/blob/2f79053d7bc7c9c9561a30dda202b3dcd2b72b90/test/escapeRegExp.test.js) by the team.
Note that I don't mean to import Lodash, just quoting (maybe with a comment for their MIT license).

References:

- [Definition of the expression to escape](https://github.com/lodash/lodash/blob/ddfd9b11a0126db2302cb70ec9973b66baec0975/lodash.js#L152)
- [Definition of the function](https://github.com/lodash/lodash/blob/ddfd9b11a0126db2302cb70ec9973b66baec0975/lodash.js#L14273-L14278)

Reply:
@NozomuIkuta Thank you for your feedback. I agree, importing lodash would be overkill. Furthermore the regex escaping needed for matching filenames is not that complex, because all it needs to do is to escape the regex reserved characters.

Could you maybe tell me how and where I would best provide a test in the code-base, so that this PR could be accepted?

Reply:
@flowrakis 

You have modified how query is set and handled.
As far as I know, this kind of logic is tested in the following 2 files:

- [packages/content/test/module.test.js](https://github.com/nuxt/content/blob/main/packages/content/test/module.test.js) to test fetching logic via [API endpoints](https://content.nuxtjs.org/advanced#api-endpoint)
- [packages/content/test/server.test.js](https://github.com/nuxt/content/blob/main/packages/content/test/server.test.js) to test fetching logic via [`$content` function](https://content.nuxtjs.org/fetching)

I guess new tests would look like ones that fetch documents with `path` parameter for API and `$content()` including characters that must be escaped, and that check documents returned are expected.

Since the escaping function is "embedded", we can't test it alone (i.e. unit test) 
If we want or need, we will have to define it as a utility function in [packages/content/lib/utils.js](https://github.com/nuxt/content/blob/main/packages/content/lib/utils.js) and create a new test file for utilities. But, IMHO, it is not needed this time.

---

Note that I'm not a core team member, but only a contributor, so what I say is based on my personal experience. But, at least I can say that test files of Nuxt Content are stored in `test` directory.

Reply:
@flowrakis 

Is there anything that I could help?